tools/registry.py

```diff
@@ -66,9 +66,10 @@ def read(path):
     return file.read()
 
 
-def integrity(data):
-  hash_value = hashlib.sha256(data)
-  return "sha256-" + base64.b64encode(hash_value.digest()).decode()
+def integrity(data, algorithm="sha256"):
+  hash = getattr(hashlib, algorithm)(data)
```

Comment:
Could you add an allowlist for algorithms that is checked before the `getattr` call? We definitely don't want `md5` or `new`.